### ISSUE TITLE
Cron task error when using Oracle DB

### ISSUE THREAD

**Issue by tquenan** [remotelearner/moodle-block_grade_me#21]
The following query from the lib.php produces a "missing keyword" error when running against an Oracle database, because 'user' is a reserved word in Oracle.

  ```
270         $sqlcourse = "SELECT count(enrol.id)
    271                       FROM {user_enrolments} enrol
    272                       LEFT JOIN {user} user ON enrol.userid = user.id
    273                       LEFT JOIN {enrol} en ON enrol.enrolid = en.id
    274                       WHERE en.courseid = ?
    275                       AND user.deleted = 0";
```

Would suggest changing the 'user' reference to just 'u' or 'usr' to avoid the issue.

**Comment by s-cenni** [comment]
I'm using Oracle and, without this fix, the {block_grade_me} table was empty and the block didn't show anything. Now it seems to work :)
Thank you @tquenan !

**Comment by jamesmcq** [comment]
Thanks @tquenan - I've scheduled this internally so stay tuned. Appreciate the PR!

**Comment by scottmacpherson** [comment]
This also affects users on PostgreSQL. Can confirm that making the changes proposed by @tquenan will resolve this bug for these users.

**Comment by ghost** [comment]
This issue is resolved in the latest releases v3.5.0.4, v3.7.0.2, and v3.9.0.2
